pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -19,40 +19,58 @@ package sidecarmetricscollector
 import (
 	"os"
 	"path/filepath"
-	"reflect"
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
 	commonv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/common/v1beta1"
 	v1beta1 "github.com/kubeflow/katib/pkg/apis/manager/v1beta1"
 	"github.com/kubeflow/katib/pkg/controller.v1beta1/consts"
 )
 
-var testJsonDataPath = filepath.Join("testdata", "JSON")
+const (
+	validJSONTestFile            = "good.json"
+	invalidFormatJSONTestFile    = "invalid-format.json"
+	invalidTimestampJSONTestFile = "invalid-timestamp.json"
+	missingMetricJSONTestFile    = "missing-objective-metric.json"
 
-func TestCollectObservationLog(t *testing.T) {
+	validTEXTTestFile            = "good.log"
+	invalidValueTEXTTestFile     = "invalid-value.log"
+	invalidFormatTEXTTestFile    = "invalid-format.log"
+	invalidTimestampTEXTTestFile = "invalid-timestamp.log"
+	missingMetricTEXTTestFile    = "missing-objective-metric.log"
+)
 
-	if err := generateTestFiles(); err != nil {
+var (
+	testDir          = "testdata"
+	testJsonDataPath = filepath.Join(testDir, "JSON")
+	testTextDataPath = filepath.Join(testDir, "TEXT")
```

Comment:
TBH, we should use the `TempDir`, but I'm ok with another PR since this static directory isn't introduced in this PR: https://pkg.go.dev/testing#T.TempDir

Please let me know what you think.

Reply:
Okay, I'll use `TempDir` in the next PR. So my work will be:
- (PR2) fix: replace static directory with TempDir.
- (PR3) chore: expose and compare errors in `file-metricscollector_test.go`

What's your opinion on it? @tenzen-y @andreyvelich

Reply:
I'd prefer to consolidate PR2 and PR3 into 1PR since we can say that "Clean up UTs".

Reply:
The motivation of both efforts (PR2 and PR3) are cleaning up metrics-collector UTs.

Reply:
Okay, that will be: 
- (PR2): fix: clean up UTs for file metrics collector

Am I right?

Reply:
I'm willing to finish that in the next few days, hopefully before `2024.3.14 12:00:00.000 GMT+8`😎.

Reply:
Correct

Reply:
I agree with @tenzen-y, we can use tmp dir here. That will also allow us to cleanup `.gitignore` file: https://github.com/kubeflow/katib/blob/master/.gitignore#L25

And as @tenzen-y said we can make this change in the following PRs, don't worry about any timeline since we can always track improvements in the issues and this is not blocking issue for the release: https://github.com/kubeflow/katib/issues/2255!

@tenzen-y @Electronic-Waste Do you want to make the changes in this PR or in the followup PRs ?

Reply:
> Do you want to make the changes in this PR or in the followup PRs ?

We'd like to work on those in a single followup PR.
Anyway, we can merge this PR now.

/hold cancel
/lgtm
/approve